### PR TITLE
[V7-994] Parse remote path information at upload time

### PR DESCRIPTION
**Pull request by andreaazzini**

### CODE REVIEW

File: darwin/importer/importer.py [hunk 1]
```diff
@@ -81,7 +81,8 @@ def import_annotations(
     print("Fetching remote file list...")
     remote_files = get_remote_files(dataset, filenames)
     for parsed_file in parsed_files:
-        if parsed_file.filename not in remote_files:
+        filename = f"{parsed_file.remote_path}/{parsed_file.filename}"
+        if filename not in remote_files:
```
Comment (simedw) [inline darwin/importer/importer.py:85]:
```suggestion
        if parsed_file.full_path not in remote_files:
```

File: darwin/item.py [hunk 2]
```diff
@@ -13,6 +13,12 @@ class DatasetItem:
     dataset_slug: str
     seq: int
     current_workflow_id: Optional[int]
+    path: str
+
+    @property
+    def full_path(self) -> str:
+        path_items = filter(lambda item: item is not None, [self.path, self.filename])
+        return "/".join(path_items)
```
Comment (simedw) [inline darwin/item.py:21]:
```suggestion
        return "/".join(filter(None, [self.path, self.filename])) 
```

File: darwin/datatypes.py [hunk 3]
```diff
@@ -77,6 +77,12 @@ class AnnotationFile:
     workview_url: Optional[str] = None
     seq: Optional[int] = None
     frame_urls: Optional[List[str]] = None
+    remote_path: Optional[str] = None
+
+    @property
+    def full_path(self) -> str:
+        path_items = filter(lambda item: item is not None, [self.remote_path, self.filename])
+        return "/".join(path_items)
```
Comment (simedw) [inline darwin/datatypes.py:85]:
```suggestion
        return "/".join(filter(None, [self.path, self.filename]))
```

File: darwin/item.py [hunk 4]
```diff
@@ -26,4 +31,5 @@ def parse_dataset_item(raw) -> DatasetItem:
         "n/a",
         raw["seq"],
         raw.get("current_workflow_id"),
+        raw["path"],
```
Comment (andreaazzini) [inline darwin/item.py:34]: @simedw is path always going to be there?

Comment (simedw) [reply darwin/item.py:34]: I think we always give it in our current api / export format

**Review by simedw — APPROVED** [review]

**Pull request merged**